core/org.osate.xtext.aadl2/src/org/osate/xtext/aadl2/validation/Aadl2JavaValidator.java

```diff
@@ -6259,7 +6261,7 @@ private void checkAccessConnectionProvidesRequires(AccessConnection connection)
 
 		// Test for L5: connection between access features of sibling components
 		if (srcContext instanceof Subcomponent && dstContext instanceof Subcomponent && source instanceof Access
-				&& destination instanceof Access) {
+				&& destination instanceof Access && destinationType != null) {
```

Comment:
There is a better way of doing this instead of a null check. Since `AccessType` is an enumeration, we can use `==` instead of `.equals()` to compare the enum literals. Use `==` for the comparisons of `sourceType` and `destinationType` to `AccessType.PROVIDES`.